### PR TITLE
cpu/nrf51: model kconfig 

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description
This models the Kconfig modules for all boards based on nrf51. The CI is set to test only this on `microbit` for now. Some changes on drivers are also included here (mostly adding hardware features to indicate presence).

### Testing procedure

- Green CI
- Check the modelling and compare `make info-modules` list with and without `TEST_KCONFIG=1`, they should match.

### Issues/PRs references

Part of #16875

### CODE REVIEW

**Comment by leandrolanzieri** [comment]
@fjmolinas thanks for tackling this one! Do you think you could add a `REMOVEME` commit like this one https://github.com/RIOT-OS/RIOT/pull/17270/commits/dd8c2b739862d676d2642c6efc51ffddfe8087c6 to quickly check all boards?

File: boards/common/microbit/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config MODULE_MICROBIT
+    bool "Microbit Support Module"
```
Comment (leandrolanzieri) [inline boards/common/microbit/Kconfig:8]: Does this one have to be visible? Should it `default y`?

Comment (fjmolinas) [reply boards/common/microbit/Kconfig:8]: Yes its what enables the `microbit` led matrix-related code, Ill change the message

File: boards/common/microbit/Kconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config MODULE_MICROBIT
+    bool "BBC micro:bit specific LED matrix handling"
+    select XTIMER
+    select MINEPLEX
```
Comment (leandrolanzieri) [inline boards/common/microbit/Kconfig:10]:
```suggestion
    select MODULE_XTIMER
    select MODULE_MINEPLEX
```

Comment (aabadie) [reply boards/common/microbit/Kconfig]: According to Makefile.dep, it should rather be ztimer/ztimer_usec

File: boards/common/microbit/Kconfig [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config MODULE_MICROBIT
+    bool "BBC micro:bit specific LED matrix handling"
+    select MODULE_XTIMER
```
Comment (aabadie) [inline boards/common/microbit/Kconfig:9]:
```suggestion
    select MODULE_ZTIMER
    select MODULE_ZTIMER_USEC
```

Comment (fjmolinas) [reply boards/common/microbit/Kconfig:9]: Hmm this was an old branch, I must have missed updating that thanks!

File: boards/calliope-mini/Makefile.dep [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ifneq (,$(filter saul_default,$(USEMODULE)))
-  USEMODULE += saul_gpio
   USEMODULE += bmx055
 endif
 
 include $(RIOTBOARD)/common/microbit/Makefile.dep
+include $(RIOTBOARD)/common/nrf51/Makefile.dep
```
Comment (fjmolinas) [inline boards/calliope-mini/Makefile.dep:6]: I think I need to split this one...

Comment (fjmolinas) [reply boards/calliope-mini/Makefile.dep:6]: Splitted and added to makefile.ci

**Comment by fjmolinas** [comment]
All green here @leandrolanzieri ?

**Comment by fjmolinas** [comment]
> All green here @leandrolanzieri ?

Shall I squash and remove the REMOVEME?

**Comment by leandrolanzieri** [comment]
Please go ahead @fjmolinas

**Review by leandrolanzieri — APPROVED** [review]
All green! ACK

**Pull request merged**